pkg/logs/tail.go

```diff
@@ -0,0 +1,610 @@
+		path string
+
+		// from is how far in the past to start showing logs.
+		from int64
+
+		// to is when to get logs until.
+		to int64
+
+		// sortBuffer is how long to buffer logs from when the cli
+		// receives them to when the cli prints them. It will sort
+		// by RequestID for that buffer period.
+		sortBuffer time.Duration
+		// searchPadding is how much of a window on either side of
+		// from and to to use for searching for the beginning or
+		// through the end timestamps.
+		searchPadding time.Duration
+		// stream specifies which of stdout or stderr or both the
+		// customer wants to consume.
+		// Undefined == both stderr and stdout.
+		stream string
+	}
+
+	// Log defines the message envelope that compute@edge (C@E) wraps the
+	// user messages in.
+	Log struct {
+		// SequenceNum is the message sequence number used to reorder
+		// messages.
+		SequenceNum int `json:"sequence_number"`
+		// RequestTime is the time in microseconds when the request
+		// was received.
+		RequestStart int64 `json:"request_start_us"`
+		// Stream is the C@E stream, either stdout or stderr.
+		Stream string `json:"stream"`
+		// RequestID is a UUID representing individual requests to the
+		// particular WASM service.
+		RequestID string `json:"id"`
+		// Message is the actual message body the user wants printed.
+		Message string `json:"message"`
+	}
+
+	// Batch encompasses a batch ID and the logs for this batch.
+	Batch struct {
+		ID   string `json:"batch_id"`
+		Logs []Log  `json:"logs"`
+	}
+)
+
+// NewTailCommand returns a usable command registered under the parent.
+func NewTailCommand(parent common.Registerer, globals *config.Data) *TailCommand {
+	var c TailCommand
+	c.Globals = globals
+	c.CmdClause = parent.Command("tail", "Tail Compute@Edge logs")
+	c.CmdClause.Flag("service-id", "Service ID").Required().StringVar(&c.Input.ServiceID)
+	c.CmdClause.Flag("from", "From time, in unix seconds").Int64Var(&c.cfg.from)
+	c.CmdClause.Flag("to", "To time, in unix seconds").Int64Var(&c.cfg.to)
+	c.CmdClause.Flag("sort-buffer",
+		"Sort buffer is how long to buffer logs before printing, defaults to 1s (second)").Default("1s").DurationVar(&c.cfg.sortBuffer)
+	c.CmdClause.Flag("search-padding",
+		"Search padding is how much of a window on either side of From and To to use for searching, defaults to 2s (seconds)").Default("2s").DurationVar(&c.cfg.searchPadding)
+	c.CmdClause.Flag("stream", "Stream specifies which of 'stdout' or 'stderr' to output, defaults to undefined (all streams)").StringVar(&c.cfg.stream)
+
+	return &c
+}
+
+// Exec invokes the application logic for the command.
+func (c *TailCommand) Exec(in io.Reader, out io.Writer) error {
+	c.Input.Kind = fastly.ManagedLoggingInstanceOutput
+	c.cfg.path = fmt.Sprintf("%s/service/%s/log_stream/managed/instance_output", config.DefaultEndpoint, c.Input.ServiceID)
+
+	c.dieCh = make(chan struct{})
+	c.batchCh = make(chan Batch)
+	c.doneCh = make(chan struct{})
+
+	c.hClient = http.DefaultClient
```

Comment:
I don't know if we want to timeout in case they actually know what they want and are requesting a window ahead of time and would like to wait on it.  Either way, this client will be going away when go-fastly has the GET logic in it.